File: app/src/main/java/com/nowfloats/util/AppsFlyerUtils.java
package com.nowfloats.util;

import android.content.Context;
import android.util.Log;

import com.appsflyer.AppsFlyerConversionListener;
import com.appsflyer.AppsFlyerLib;

import java.util.Map;

public class AppsFlyerUtils implements AppsFlyerConversionListener {
    private static final String TAG = AppsFlyerUtils.class.getName();

    public static void initAppsFlyer(Context context, String devKey) {
        AppsFlyerLib.getInstance().init(devKey, new AppsFlyerUtils(), context);
        //Start the SDK
        AppsFlyerLib.getInstance().start(context);
        //Enable Debugging
        AppsFlyerLib.getInstance().setDebugLog(true);
    }

    @Override
    public void onConversionDataSuccess(Map<String, Object> conversionData) {
        for (String attrName : conversionData.keySet()) {
            Log.d(TAG, "attribute: " + attrName + " = " + conversionData.get(attrName));
        }
    }

    @Override
    public void onConversionDataFail(String errorMessage) {
        Log.d(TAG, "error getting conversion data: " + errorMessage);
    }

    @Override
    public void onAppOpenAttribution(Map<String, String> attributionData) {
        for (String attrName : attributionData.keySet()) {
            Log.d(TAG, "attribute: " + attrName + " = " + attributionData.get(attrName));
        }
    }

    @Override
    public void onAttributionFailure(String errorMessage) {
        Log.d(TAG, "error onAttributionFailure : " + errorMessage);
    }
}

File: framework/src/main/java/com/framework/utils/AppsFlyerUtils.kt
package com.framework.utils

import android.content.Context
import android.util.Log
import com.appsflyer.AppsFlyerConversionListener
import com.appsflyer.AppsFlyerLib

class AppsFlyerUtils : AppsFlyerConversionListener {

  companion object {
    private val TAG = AppsFlyerUtils::class.java.name

    @JvmField
    var sAttributionData: Map<String, String> = mapOf()

    @JvmStatic
    fun initAppsFlyer(context: Context?, devKey: String?) {
      AppsFlyerLib.getInstance().init(devKey!!, AppsFlyerUtils(), context!!)
      //Start the SDK
      AppsFlyerLib.getInstance().start(context)
      //Enable Debugging
      AppsFlyerLib.getInstance().setDebugLog(true)
    }

  }

  override fun onConversionDataSuccess(conversionData: Map<String, Any>) {
    for (attrName in conversionData.keys) {
      Log.d(TAG, "attribute: " + attrName + " = " + conversionData[attrName])
    }
  }

  override fun onConversionDataFail(errorMessage: String) {
    Log.d(TAG, "error getting conversion data: $errorMessage")
  }

  override fun onAppOpenAttribution(attributionData: Map<String, String>) {
    sAttributionData = attributionData
    for (attrName in attributionData.keys) {
      Log.d(TAG, "attribute: " + attrName + " = " + attributionData[attrName])
    }
  }

  override fun onAttributionFailure(errorMessage: String) {
    Log.d(TAG, "error onAttributionFailure : $errorMessage")
  }
}